spec/system/donation_system_spec.rb

```diff
@@ -418,7 +418,7 @@
             expect(page).to have_xpath("//select[@id='donation_line_items_attributes_0_item_id']")
           end.not_to change { Donation.count }
           expect(page).to have_content("Start a new donation")
-          expect(page).to have_content("must be less than")
+          expect(page).to have_content("Quantity is not a number - Note: commas are not allowed")
```

Comment:
um... why? This is a number, isn't it?

Reply:
- Line 412 is setting the quantity - `fill_in "donation_line_items_attributes_0_quantity", with: "10000000000000000000000"` 
  - This number is bigger than the configured limit of `2**31 = 2,147,483,648`

Reply:
Which means it should be showing the 'must be less than',  rather than the "Quantity is not a number"

Reply:
We're checking for non-numeric characters in the unit tests. Do I need to do that for the system tests as well?

Reply:
No - it's just a weird error message since I'd expect to test a string like "abc" here instead of a number that's too large.

Reply:
i.e. let's make the tests explicit that the input is a number with commas rather than relying on implicit changes that happen in the controller / http layer.

Reply:
I'm a little confused.    The original test was about overshooting the maximum.  @vsnrth is changing the meaning of the test, leaving overshooting the maximum  untested....  We should still have that test,  no?

Reply:
Yes - we need both tests (one where it doesn't get a number and one where the number is too high).

Reply:
Updated the code to handle the request. We get a different message when the limit is crossed and when a string is entered.